std/concurrency.d

```diff
@@ -1076,6 +1076,24 @@ Tid locate( string name )
     }
 }
 
+/**
+ * Gets all of the names associated with the $(LREF Tid).
+ *
+ * Params:
+ *     tid = The $(LREF Tid) to locate within the registry.
+ *
+ * Returns:
+ *     The associated name or `null` if `tid` is not registered.
+ */
+string[] locate(Tid tid)
+{
+    synchronized (registryLock)
+    {
+        if (auto name = tid in namesByTid)
+            return *name;
+        return null;
+    }
```

Comment:
suggestion: replace the whole implementation with `synchronized return namesByTid.get(tid, null);`